components/shared-image-gallery/image_gallery.tf

```diff
@@ -6,38 +6,48 @@ resource "azurerm_shared_image_gallery" "image_gallery" {
   tags = module.ctags.common_tags
 }
 
-resource "azurerm_shared_image" "bastion-ubuntu" {
-  name                = var.image_name
+resource "azurerm_shared_image" "shared_image" {
+  for_each            = var.images
+  name                = each.value.name
   gallery_name        = azurerm_shared_image_gallery.image_gallery.name
   resource_group_name = azurerm_resource_group.image_gallery_rg.name
   location            = azurerm_resource_group.image_gallery_rg.location
   os_type             = "Linux"
 
   identifier {
     publisher = "hmcts"
-    offer     = var.image_name
-    sku       = "18.04-LTS"
+    offer     = each.value.offer
+    sku       = each.value.sku
   }
 }
 
-data "azurerm_image" "bastion-ubuntu" {
-  name                = "rdo-bastion-image-151746"
-  resource_group_name = "rdo-bastion-packer"
+resource "azurerm_managed_disk" "managed_disk" {
```

Comment:
do these cost us anything leaving them lying around?

Reply:
It's in the order of pence per month according to https://azure.microsoft.com/en-gb/pricing/details/managed-disks/

I've updated it to be smaller since it's only being used to create a base image version.

I will look again at whether it can be deleted in the pipeline. I was previously relying on creating a VM to accomplish the same effect but deleting that kept deleting the image versions along with it and this involves much less redundant infrastructure.

Reply:
Deleting the managed disk causes the image version to be deleted as well due to the dependencies. I've reduced the size so cost should be minimal.